codegen/src/api/mod.rs

```diff
@@ -222,76 +223,55 @@ impl RuntimeGenerator {
                 #( #modules )*
                 #types_mod
 
-                /// Default configuration of common types for a target Substrate runtime.
-                #[derive(Clone, Debug, Default, Eq, PartialEq)]
-                pub struct DefaultConfig;
-
-                impl ::subxt::Config for DefaultConfig {
-                    type Index = u32;
-                    type BlockNumber = u32;
-                    type Hash = ::subxt::sp_core::H256;
-                    type Hashing = ::subxt::sp_runtime::traits::BlakeTwo256;
-                    type AccountId = ::subxt::sp_runtime::AccountId32;
-                    type Address = ::subxt::sp_runtime::MultiAddress<Self::AccountId, u32>;
-                    type Header = ::subxt::sp_runtime::generic::Header<
-                        Self::BlockNumber, ::subxt::sp_runtime::traits::BlakeTwo256
-                    >;
-                    type Signature = ::subxt::sp_runtime::MultiSignature;
-                    type Extrinsic = ::subxt::sp_runtime::OpaqueExtrinsic;
-                }
-
-                impl ::subxt::ExtrinsicExtraData<DefaultConfig> for DefaultConfig {
-                    type AccountData = AccountData;
-                    type Extra = ::subxt::DefaultExtra<DefaultConfig>;
-                }
+                /// The default storage entry from which to fetch an account nonce, required for
+                /// constructing a transaction.
+                pub type DefaultAccountData = self::system::storage::Account;
 
-                pub type AccountData = self::system::storage::Account;
-
-                impl ::subxt::AccountData<DefaultConfig> for AccountData {
-                    fn nonce(result: &<Self as ::subxt::StorageEntry>::Value) -> <DefaultConfig as ::subxt::Config>::Index {
+                impl ::subxt::AccountData<::subxt::DefaultConfig> for DefaultAccountData {
+                    fn nonce(result: &<Self as ::subxt::StorageEntry>::Value) -> <::subxt::DefaultConfig as ::subxt::Config>::Index {
                         result.nonce
                     }
-                    fn storage_entry(account_id: <DefaultConfig as ::subxt::Config>::AccountId) -> Self {
+                    fn storage_entry(account_id: <::subxt::DefaultConfig as ::subxt::Config>::AccountId) -> Self {
                         Self(account_id)
                     }
                 }
 
-                pub struct RuntimeApi<T: ::subxt::Config + ::subxt::ExtrinsicExtraData<T>> {
+                pub struct RuntimeApi<T: ::subxt::Config, E> {
```

Comment:
Is it worth defaulting these generic params eg

```
pub struct RuntimeApi<T: ::subxt::Config = ::subxt::DefaultConfig, E = ::subxt::DefaultExtra<::subxt::DefaultConfig>> {
```

so that we don't have to specify them when instantiating the `RuntimeApi` unless they differ? Or is the common case that you'll need to specify different ones, so it may as well be explicit?

Reply:
I'm not sure TBH, on the one hand it is very convenient if you are using the defaults, on the other if they are hidden then there is not the clue that it is configurable.

I'd lean towards leaving it explicit for now, until the api has solidified and we have an idea of the common usage.